compiler/accelerators/snax.py

```diff
@@ -112,16 +112,21 @@ def _generate_streamer_setup_vals(
         result: Sequence[tuple[Sequence[Operation], SSAValue]] = []
 
         # loop bound registers
-        loop_bounds: Sequence[IntAttr] = op.stride_patterns.data[0].upper_bounds.data
-        result.extend(
-            [
-                (
-                    [cst := arith.Constant.from_int_and_width(loop_bound.data, i32)],
-                    cst.result,
-                )
-                for loop_bound in loop_bounds
-            ]
-        )
+        for operand, streamer in enumerate(self.streamer_config.data.streamers):
+            upper_bounds = op.stride_patterns.data[operand].upper_bounds.data
+            while streamer.temporal_dim > len(upper_bounds):
+                # if not all temporal bounds are used, insert 1's
+                upper_bounds = (IntAttr(1),) + upper_bounds
```

Comment:
I would write this like you are writing it in the comment (if possible), not sure if the if statement is necessary anymore then:
```suggestion
           # pad unused temporal bounds with 1's
           upper_bounds = (IntAttr(1),) * (streamer.temporal_dim - len(upper_bounds))  + upper_bounds
```